### PR TITLE
Add event connections to attendees types.

### LINKED ISSUE #24: [Feature Request] Add connected event to RSVP/Tickets and Attendees
Trying to replicate the Who's Attending block, as well as display the associated event on a user's ticket, but it seems that information isnt part of either the the various ticket nodes, nor the attendee nodes.

Happy to try it myself and submit a PR, but I'm not entirely sure where in the docs (or the plugin) these sorts of connections are covered. A push in the right direction would be really appreciated!

### PR DESCRIPTION
**Pull request by justlevine**
Adds event connections to the various attendee types. Closes #24 .

1. The `Event` type has `rSVPAttendees`, `payPalAttendees`, and `wooAttendees`.
2. `rSVPAttendees`, `payPalAttendees`, and `wooAttendees` can now be filtered by `where : { eventsIn: $id }`.

Note:
- Separate `class-{type}attendee.php` and `class-{type}attendee-resolver.php` files were created for each attendee type. This isn't very DRY, but was the best I could do without fully changing attendee over to an `interface`.
- The old `class-attendee-resolver.php` was never actually included before. There's [a code segment](https://github.com/simplur/ql-events/blob/66125f4a32afad5f60e0ee2c438967802e4a5bbc/includes/data/connection/class-attendee-connection-resolver.php#L37) from @kidunot89 that I left in, because I'm not really sure of the purpose/its effect. If it isn't necessary, it should be removed.
- For some reason, `composer dump-autoload` was generating `platform_check.php`. I manually removed it, but to be safe, a maintainer should probably regenerate the autoload.

### CODE REVIEW

File: includes/class-core-schema-filters.php [hunk 1]
```diff
@@ -75,6 +75,33 @@ public static function add_filters() {
...
+				10,
+				5
+			);
+
+			add_filter(
+				'graphql_post_object_connection_query_args',
+				array(
+					'\WPGraphQL\QL_Events\Data\Connection\PayPalAttendee_Connection_Resolver',
+					'get_query_args',
+				),
```
Comment (bordoni) [inline includes/class-core-schema-filters.php:94]:
```suggestion
				[
					'\WPGraphQL\QL_Events\Data\Connection\PayPalAttendee_Connection_Resolver',
					'get_query_args',
				],
```

File: includes/class-core-schema-filters.php [hunk 2]
```diff
@@ -75,6 +75,33 @@ public static function add_filters() {
...
+				10,
+				5
+			);
+
+			add_filter(
+			'graphql_input_fields',
+			array( __CLASS__, 'attendees_where_args' ),
+			10,
+			2
+		);
```
Comment (bordoni) [inline includes/class-core-schema-filters.php:104]:
```suggestion
			add_filter( 'graphql_input_fields', [ __CLASS__, 'attendees_where_args' ], 10, 2 );
```

File: includes/class-core-schema-filters.php [hunk 3]
```diff
@@ -97,6 +124,23 @@ public static function add_filters() {
 				10,
 				6
 			);
+
+			add_filter(
+				'graphql_post_object_connection_query_args',
+				array(
+					'\WPGraphQL\QL_Events\Data\Connection\WooAttendee_Connection_Resolver',
+					'get_query_args',
+				),
```
Comment (bordoni) [inline includes/class-core-schema-filters.php:133]:
```suggestion
				[
					'\WPGraphQL\QL_Events\Data\Connection\WooAttendee_Connection_Resolver',
					'get_query_args',
				],
```

File: includes/class-core-schema-filters.php [hunk 4]
```diff
@@ -97,6 +124,23 @@ public static function add_filters() {
...
+				10,
+				5
+			);
+
+			add_filter(
+				'graphql_input_fields',
+				array( __CLASS__, 'ticket_plus_attendees_where_args' ),
```
Comment (bordoni) [inline includes/class-core-schema-filters.php:140]:
```suggestion
				[ __CLASS__, 'ticket_plus_attendees_where_args' ],
```

File: includes/connection/class-paypalattendees.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	public static function register_connections() {
+		// From Event to Attendees.
+		register_graphql_connection(
+			self::get_connection_config(
+				get_post_type_object( PAYPAL::ATTENDEE_OBJECT ),
+				array(
+					'fromType'      => 'Event',
+					'toType'        => 'PayPalAttendee',
+					'fromFieldName' => 'payPalAttendees',
+				)
```
Comment (bordoni) [inline includes/connection/class-paypalattendees.php:32]:
```suggestion
				[
					'fromType'      => 'Event',
					'toType'        => 'PayPalAttendee',
					'fromFieldName' => 'payPalAttendees',
				]
```

File: includes/connection/class-rsvpattendees.php [hunk 6]
```diff
@@ -27,10 +27,19 @@ public static function register_connections() {
 				get_post_type_object( RSVP::ATTENDEE_OBJECT ),
 				array(
 					'fromType'      => 'Event',
-					'toType'        => 'Attendee',
-					'fromFieldName' => 'attendees',
+					'toType'        => 'RSVPAttendee',
+					'fromFieldName' => 'rSVPAttendees',
```
Comment (bordoni) [inline includes/connection/class-rsvpattendees.php:31]: Is this `r` lower case intentional?

File: includes/connection/class-wooattendees.php [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	public static function register_connections() {
+		// From Event to Attendees.
+		register_graphql_connection(
+			self::get_connection_config(
+				get_post_type_object( WOO::ATTENDEE_OBJECT ),
+				array(
+					'fromType'      => 'Event',
+					'toType'        => 'WooAttendee',
+					'fromFieldName' => 'wooAttendees',
+				)
```
Comment (bordoni) [inline includes/connection/class-wooattendees.php:32]:
```suggestion
				[
					'fromType'      => 'Event',
					'toType'        => 'WooAttendee',
					'fromFieldName' => 'wooAttendees',
				]
```

File: includes/connection/class-wooattendees.php [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+		);
+	}
+
+	public static function where_args() {
+		return array(
+			'eventsIn' => array(
+				'type' => array( 'list_of' => 'ID'),
+				'description' => __( 'Filter the connection based on event Id'),
+			)
+		);
```
Comment (bordoni) [inline includes/connection/class-wooattendees.php:43]:
```suggestion
		return [
			'eventsIn' => [
				'type' => [ 'list_of' => 'ID' ],
				'description' => __( 'Filter the connection based on event Id', 'ql-events' ),
			]
		];
```

File: includes/data/connection/class-paypalattendee-connection-resolver.php [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+					// @codingStandardsIgnoreLine
+					if ( 'payPalAttendees' === $info->fieldName ) {
+						if ( ! isset( $query_args['meta_query'] ) ) {
+							$query_args['meta_query'] = array(); // WPCS: slow query ok.
+						}
+						$query_args['meta_query'][] = array(
+							'key'     => PAYPAL::ATTENDEE_EVENT_KEY,
+							'value'   => $source->ID,
+							'compare' => '=',
+						);
```
Comment (bordoni) [inline includes/data/connection/class-paypalattendee-connection-resolver.php:64]:
```suggestion
						$query_args['meta_query'][] = [
							'key'     => PAYPAL::ATTENDEE_EVENT_KEY,
							'value'   => $source->ID,
							'compare' => '=',
						];
```

File: includes/data/connection/class-paypalattendee-connection-resolver.php [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+		// Determine where we're at in the Graph and adjust the query context appropriately.
+		// @Todo: this was left over from class-attendee-connection-resolver.php which wasnt loaded. May not be necessary.
+		if ( true === is_object( $source ) ) {
+
+			switch ( $source->post_type ) {
+				case Main::POSTTYPE:
+					// @codingStandardsIgnoreLine
+					if ( 'payPalAttendees' === $info->fieldName ) {
+						if ( ! isset( $query_args['meta_query'] ) ) {
+							$query_args['meta_query'] = array(); // WPCS: slow query ok.
```
Comment (bordoni) [inline includes/data/connection/class-paypalattendee-connection-resolver.php:58]:
```suggestion
							$query_args['meta_query'] = []; // WPCS: slow query ok.
```

File: includes/connection/class-rsvpattendees.php [hunk 11]
```diff
@@ -27,10 +27,19 @@ public static function register_connections() {
...
 		);
 	}
+
+	public static function where_args() {
+		return array(
+			'eventsIn' => array(
+				'type' => array( 'list_of' => 'ID'),
+				'description' => __( 'Filter the connection based on event Id'),
+			)
+		);
```
Comment (bordoni) [inline includes/connection/class-rsvpattendees.php:43]:
```suggestion
		return [
			'eventsIn' => [
				'type' => [ 'list_of' => 'ID' ],
				'description' => __( 'Filter the connection based on event Id', 'ql-events' ),
			]
		];
```

**Review by bordoni — CHANGES_REQUESTED** [review]
Suggested a couple small things.

**Comment by justlevine** [comment]
Requested changes merged.

**Comment by kidunot89** [comment]
@justlevine @bordoni This functionality was implemented in #46

**Pull request closed without merging**